CVLByExample/ERC20/certora/specs/ERC20Fixed.spec

```diff
@@ -117,3 +123,88 @@ rule sanity {
   f(e, arg);
   satisfy true;
 }
+
+
+// satisfy examples
+// Generate an example trace for a first deposit operation that succeeds.
+rule satisfyFirstDepositSucceeds(){
+    env e;
+    require totalSupply() == 0;
+    deposit(e);
+    satisfy totalSupply() == e.msg.value;
+}
+
+// Generate an example trace for a withdraw that results totalSupply == 0.
+rule satisfyLastWithdrawSucceeds() {
+    env e;
+    uint256 amount;
+    requireInvariant totalSupplyIsSumOfBalances();
+    require totalSupply() > 0;
+    withdraw(e, amount);
+    satisfy totalSupply() == 0;
+}
+
+
+rule satisfyWithManyOps(){
+    env e; address recipient; uint amount;
+
+    require balanceOf(e.msg.sender) > amount;
+    require e.msg.value == 0;
+    require balanceOf(recipient) + amount < max_uint;
+    require e.msg.sender != 0;
+    require recipient != 0;
+    deposit(e);
+    depositAmount(e, amount);
+    transfer@withrevert(e, recipient, amount);
+    satisfy totalSupply() > 0;  
+}
+
+
+
+// A non-vacuous example where transfer() does not revert.
+rule satisfyVacuityCorrection {
+    env e; address recip; uint amount;
+
+    require balanceOf(e.msg.sender) > 0;
+
+    transfer@withrevert(e, recip, amount);
+
+    satisfy balanceOf(e.msg.sender) == 0;
```

Comment:
to show a case that transfer does not revert remove the withrevert

Reply:
removed